forte/data/data_store.py

```diff
@@ -1282,23 +1325,23 @@ def co_iterator_annotation_like(
             # `the current entry` means the entry that
             #      popped entry_tuple represents.
             # `the current entry list` means the entry
-            # list (values of self.__elements) where `the current entry`
+            # list (values of all_entries) where `the current entry`
             # locates at.
 
             # retrieve the popped entry tuple (minimum item in the heap)
-            # and get the p_idx (the index of the current entry list in self.__elements)
+            # and get the p_idx (the index of the current entry list in all_entries)
             entry_tuple = heappop(h)
             (_, _, p_idx), type_name = entry_tuple
             # get the index of current entry
             # and locate the entry represented by the tuple for yielding
             pointer = pointers[p_idx]
-            entry = self.__elements[type_name][pointer]
+            entry = all_entries[type_name][pointer]
```

Comment:
As suggested above, you can adjust the `pointer` here to the `begin_index`, and the line below to be `pointer + 1 < end_index`. Of course, you need to maintain different `begin_index` and `end_index` for all types.